CI: use different TMPDIR on prior-fedora

edsantiago opened the pull request:
A small number of tests are assuming that TMPDIR == /tmp. These tests fail when that assumption does not hold.

Set TMPDIR=/var/tmp on prior-fedora, as a way to catch those. /dev/shm would be a slightly better choice, because the string "tmp" does not appear it in, but it's way too small to be of any use: it fills up in the e2e prefetch.

This PR exposed a nasty bug in our Makefile: using "TMPDIR" as temporary variable completely unrelated to (and inconsistent with) the actual established use of TMPDIR. Solution: rename that variable and make it lower case. Do the same with two other ALL-CAP variables.

Signed-off-by: Ed Santiago <santiago@redhat.com>
```release-note
None
```

edsantiago commented:
This is going to fail CI, because #19267 has not merged.

Luap99 commented:
You PR description says: `TMPDIR=/dev/shm` while the commit and code uses `/var/tmp`.
Did you try /dev/shm first and that failed?

Also did you check the test speed? If it no longer uses tmpfs it may be slower.

edsantiago commented:
`/dev/shm` was my first try. It failed (because of the broken Makefile:TMPDIR stuff, but I didn't know that).  I updated to `/var/tmp`, changed it in the git commit message, but not in Cirrus because WIP. Only after that did I discover the broken Makefile.

Tmpfs: not a factor. I did a Rerun-in-terminal and noticed that `/tmp` is not tmpfs, it's just a subdir of root. `/dev/shm` is a 2G tmpfs. I'm undecided as to whether to try to go back to it, if that will be enough space for e2e tests.

Speed: `cirrus-pr-timing` actually shows a very slight speedup, O(1m), but that's well within normal variation so I consider it an artifact:

| type | distro              | user     | DB     |    local |   remote | container |
| ---- | --------            | ----     | --     |    ----: |   -----: | --------: |
| 17831 (baseline) |
| int  | fedora-37           | root     | sqlite |    38:27 |    39:57 |     38:31 |
| int  | fedora-37           | rootless | sqlite |    34:34 |          |
| sys  | fedora-37           | root     | sqlite |    36:24 |    22:29 |
| sys  | fedora-37           | rootless | sqlite |    33:35 |          |
| 19281 (this PR) |
| int  | fedora-37           | root     |        |    37:29 |    38:13 |     37:48 |
| int  | fedora-37           | rootless |        |    32:46 |          |
| sys  | fedora-37           | root     |        |   !34:01 |    21:20 |
| sys  | fedora-37           | rootless |        |    31:44 |          |

Luap99 commented on .cirrus.yml:
```diff
@@ -123,6 +123,7 @@ build_task:
               CTR_FQIN: ${PRIOR_FEDORA_CONTAINER_FQIN}
               CI_DESIRED_RUNTIME: crun
               CI_DESIRED_NETWORK: cni
+              TMPDIR: /var/tmp
```
In case you have to repush please add a comment why we are doing this. I know it is in the git history but once it get moved it will be much harder to figure out at some point in the future.

edsantiago replied:
I meant to do that, thanks. Will re-push once I see how CI does.

Luap99 reviewed (commented):
LGTM

edsantiago commented:
@containers/podman-maintainers could we get this in? The bogus-TMPDIR bug led to wasted energy on #19351, and will no doubt cause other future problems.

rhatdan commented:
/lgtm

edsantiago commented:
/hold

because CI hasn't finished

edsantiago commented:
/hold cancel
